common/djangoapps/django_sudo_helpers/pipeline.py

```diff
@@ -0,0 +1,38 @@
+"""django-sudo pipeline definitions."""
+
+import re
+
+from django.conf import settings
+from social.pipeline import partial
+
+from sudo.utils import grant_sudo_privileges
+from util.request import COURSE_REGEX
+
+AUTH_ENTRY_SUDO = 'grant_sudo'
+
+# pylint: disable=unused-argument
+
+
+@partial.partial
+def create_sudo_session(strategy, auth_entry, *args, **kwargs):
+    """
+    Grant sudo privileges if user is authenticated and auth_entry
+    is equal to AUTH_ENTRY_SUDO.
+    """
+    request = strategy.request if strategy else None
+    if auth_entry == AUTH_ENTRY_SUDO and request.user.is_authenticated():
+        _region = None
+        redirect_to = strategy.session_get('next', '')
+        match = COURSE_REGEX.match(redirect_to)
+        if not match:
+            # COURSE_REGEX will not work for studio
+            course_regex = re.compile(r'^.*?/course_team/{}'.format(settings.COURSE_ID_PATTERN))
```

Comment:
@dan-f I don't think it will work because we are matching ```course_id``` from url, consider this if url is like this ```/course_team/edx/demo/2015_Q2``` and we match against ```settings.COURSE_ID_PATTERN``` then it will return this ```course_team/edx/demo``` instead of ```edx/demo/2015_Q2``` thats why I'm including ```/course_team/``` in regex like ```/courses/``` in ```COURSE_REGEX```.